site/src/contexts/auth/useAuth.ts

```diff
@@ -10,3 +10,22 @@ export const useAuth = () => {
 
   return context;
 };
+
+
+
+export const useAuthenticated = () => {
+  const auth = useAuth()
+
+  if(!auth.user) {
+    throw new Error("User is not authenticated.")
+  }
+
+  // We can do some TS magic here but I would rather to be explicit on what
+  // values are not undefined when authenticated
+  return auth as AuthContextValue & {
+    user: Exclude<AuthContextValue['user'], undefined>,
+    permissions: Exclude<AuthContextValue['permissions'], undefined>,
+    orgId: Exclude<AuthContextValue['orgId'], undefined>,
+  }
```

Comment:
I think this type assertion is silencing a possible race condition with `permissions`

`permissions` comes from `permissionsQuery`, which is dependent on `userQuery`. But, since `permissionsQuery` has to wait for `userQuery` to finish before it can start running, there's going to be at least some period of time when we have the user data but not the permissions data

Could we do two runtime checks – one on `auth.user` and one on `permissions`?